modules/core/src/main/scala/lucuma/odb/api/schema/ScienceRequirementsSchema.scala

```diff
@@ -0,0 +1,157 @@
+          fieldType   = BigDecimalType,
+          description = Some("Focal plane angle in mas"),
+          resolve     = v => Angle.signedDecimalMilliarcseconds.get(v.value)
+        ),
+
+        Field(
+          name        = "arcseconds",
+          fieldType   = BigDecimalType,
+          description = Some("Focal plane angle in arcsec"),
+          resolve     = v => Angle.signedDecimalArcseconds.get(v.value)
+        )
+
+      )
+    )
+
+  def SpectroscopyRequirementsType[F[_]]: ObjectType[OdbRepo[F], SpectroscopyScienceRequirements] =
+    ObjectType(
+      name     = "SpectroscopyScienceRequirements",
+      fieldsFn = () =>
+        fields(
+          Field(
+            name        = "wavelength",
+            fieldType   = OptionType(WavelengthType[F]),
+            description = Some("Requested central wavelength"),
+            resolve     = _.value.wavelength
+          ),
+
+          Field(
+            name        = "resolution",
+            fieldType   = OptionType(IntType),
+            description = Some("Requested resolution"),
+            resolve     = _.value.resolution.map(_.value)
+          ),
+
+          Field(
+            name        = "signalToNoise",
+            fieldType   = OptionType(BigDecimalType),
+            description = Some("Requested signal to noise ratio"),
+            resolve     = _.value.signalToNoise.map(_.value)
+          ),
+
+          Field(
+            name        = "signalToNoiseAt",
+            fieldType   = OptionType(WavelengthType[F]),
+            description = Some("Requested wavelength for the requested signal to noise"),
+            resolve     = _.value.signalToNoiseAt
+          ),
+
+          Field(
+            name        = "wavelengthRange",
+            fieldType   = OptionType(WavelengthType[F]),
+            description = Some("Wavelength range"),
+            resolve     = _.value.wavelengthRange
+          ),
+
+          Field(
+            name        = "focalPlane",
+            fieldType   = OptionType(EnumTypeFocalPlane),
+            description = Some("Focal plane choice"),
+            resolve     = _.value.focalPlane
+          ),
+
+          Field(
+            name        = "focalPlaneAngle",
+            fieldType   = OptionType(FocalPlaneAngleType[F]),
+            description = Some("Focal plane angle"),
+            resolve     = _.value.focalPlaneAngle
+          ),
+
+          Field(
+            name        = "capabilities",
+            fieldType   = OptionType(EnumTypeSpectroscopyCapabilities),
+            description = Some("Spectroscopy Capabilities"),
+            resolve     = _.value.capabilities
+          ),
+
+        )
+    )
+
+  def ScienceRequirementsType[F[_]]: ObjectType[OdbRepo[F], ScienceRequirements] =
+    ObjectType(
+      name     = "ScienceRequirements",
+      fieldsFn = () =>
+        fields(
+          Field(
+            name        = "mode",
+            fieldType   = EnumTypeScienceMode,
```

Comment:
Should these be `OptionType`?

Reply:
Only if it is indeed optional in the model itself.  Here, as currently defined, the mode is always present so it will always be present if this field is included in the query.